spec/date.html

```diff
@@ -831,6 +838,9 @@ <h1>CompareTemporalDate ( _one_, _two_ )</h1>
         1. If _one_.[[ISOMonth]] &lt; _two_.[[ISOMonth]], return -1.
         1. If _one_.[[ISODay]] &gt; _two_.[[ISODay]], return 1.
         1. If _one_.[[ISODay]] &lt; _two_.[[ISODay]], return -1.
+        1. <mark>TODO: make sure this is correct logic.</mark>
+        1. If _one_.[[Calendar]].id &gt; _two_.[[Calendar]].id, return 1.
+        1. If _one_.[[Calendar]].id &lt; _two_.[[Calendar]].id, return -1.
```

Comment:
`DateDifference` checks that the [[Calendar]] slots are equal in advance, so let's put this in `Date.prototype.compare` itself:

```diff
+        1. Let _result_ be ! CompareTemporalDate(_one_.[[ISOYear]], _one_.[[ISOMonth]], _one_.[[ISODay]], _two_.[[ISOYear]], _two_.[[ISOMonth]], _two_.[[ISODay]]).
+        1. If _result_ ≠ 0, then
+          1. Return _result_.
+        1. Let _calendarOne_ be ? CalendarToString(_one_.[[Calendar]]).
+        1. Let _calendarTwo_ be ? CalendarToString(_two_.[[Calendar]]).
+        1. Let _r_ be the result of performing Abstract Relational Comparison _calendarOne_ &lt; _calendarTwo_.
+        1. If _r_ is *true*, return -1.
+        1. Let _r_ be the result of performing Abstract Relational Comparison _calendarTwo_ &lt; _calendarOne_.
+        1. If _r_ is *true*, return 1.
+        1. Return +0.
```

Reply:
Done